src/Sensor/SensorOptical.cpp

```diff
@@ -32,11 +32,36 @@ void SensorOptical::begin() {
 
 void SensorOptical::handle() {
     uint32_t currentTime = micros();
-    if (currentTime >= _lastUpdateTime + _readInterval) {
+    uint32_t nextUpdateTime = _lastUpdateTime + _readInterval;
```

Comment:
I'm not sure how much computation is required if we calculate ```nextUpdateTime``` and ```overflow``` each time we call ```handle()```. If this uses too much overhead, we could keep them as ```SensorOptical``` properties, since they only change during updates.